### ISSUE TITLE
[4.y] Watching resources not working in IPv6 cluster

### ISSUE THREAD

**Issue by harshit-splunk** [ManageIQ/kubeclient#585]
In single stack IPv6-only cluster, pulling the resources work but watching any resources results in `getaddrinfo: Name or service not known (HTTP::ConnectionError)` error.

## Steps to reproduce:

```ruby
require 'kubeclient'
require 'resolv'

ssl_options = {
  verify_ssl: OpenSSL::SSL::VERIFY_NONE
}

auth_options = {}
auth_options[:bearer_token] = File.read("/var/run/secrets/kubernetes.io/serviceaccount/token")

api_endpoint = 'api'
api_version = 'v1'
kubernetes_url = ""
env_host = ENV['KUBERNETES_SERVICE_HOST']
env_host = "[#{env_host}]" if env_host =~ Resolv::IPv6::Regex
env_port = ENV['KUBERNETES_SERVICE_PORT']
if env_host && env_port
  kubernetes_url = "https://#{env_host}:#{env_port}/api"
end
puts kubernetes_url
client = Kubeclient::Client.new(
  kubernetes_url, api_version,
  ssl_options: ssl_options,
  auth_options: auth_options
)
puts client.api_valid?
options = {}
options[:as] = :raw
pods = client.public_send("get_pods", options)
pods = JSON.parse(pods)
puts pods.size
watcher = client.public_send("watch_pods", options)
watcher.each do |entity|
  entity = JSON.parse(entity)
  puts entity.size
end
```

## Output:

```
bash-4.4$ bundle exec ruby /tmp/demo.rb 
https://[fd62:3165:1d16::1]:443/api
true
4
Traceback (most recent call last):
        8: from /tmp/demo.rb:33:in `<main>'
        7: from /usr/share/gems/bundler/gems/kubeclient-955ec5b62f64/lib/kubeclient/watch_stream.rb:19:in `each'
        6: from /usr/share/gems/gems/http-4.4.1/lib/http/client.rb:31:in `request'
        5: from /usr/share/gems/gems/http-4.4.1/lib/http/client.rb:71:in `perform'
        4: from /usr/share/gems/gems/http-4.4.1/lib/http/client.rb:71:in `new'
        3: from /usr/share/gems/gems/http-4.4.1/lib/http/connection.rb:43:in `initialize'
        2: from /usr/share/gems/gems/http-4.4.1/lib/http/timeout/null.rb:21:in `connect'
        1: from /usr/share/gems/gems/http-4.4.1/lib/http/timeout/null.rb:21:in `open'
/usr/share/gems/gems/http-4.4.1/lib/http/timeout/null.rb:21:in `initialize': getaddrinfo: Name or service not known (SocketError)
        8: from /tmp/demo.rb:33:in `<main>'
        7: from /usr/share/gems/bundler/gems/kubeclient-955ec5b62f64/lib/kubeclient/watch_stream.rb:19:in `each'
        6: from /usr/share/gems/gems/http-4.4.1/lib/http/client.rb:31:in `request'
        5: from /usr/share/gems/gems/http-4.4.1/lib/http/client.rb:71:in `perform'
        4: from /usr/share/gems/gems/http-4.4.1/lib/http/client.rb:71:in `new'
        3: from /usr/share/gems/gems/http-4.4.1/lib/http/connection.rb:43:in `initialize'
        2: from /usr/share/gems/gems/http-4.4.1/lib/http/timeout/null.rb:21:in `connect'
        1: from /usr/share/gems/gems/http-4.4.1/lib/http/timeout/null.rb:21:in `open'
/usr/share/gems/gems/http-4.4.1/lib/http/timeout/null.rb:21:in `initialize': failed to connect: getaddrinfo: Name or service not known (HTTP::ConnectionError)
```

Kubeclient uses rest-client gem for pulling objects and http gem for watching it.
I believe the is related to http gem. There is similar issue created in their repo(https://github.com/httprb/http/issues/730)

**Comment by harshit-splunk** [comment]
```
require 'net/http'

uri=URI("#{kubernetes_url}/v1/namespaces/default/pods")
puts uri.request_uri
http = Net::HTTP.new(uri.host, uri.port)
http.use_ssl = true
http.verify_mode = OpenSSL::SSL::VERIFY_NONE

request = Net::HTTP::Get.new(uri.request_uri)
request["Authorization"] =  "Bearer #{token}"
response = http.request(request)
puts response.code
```

`net/http` gem is working successfully.  
@cben what are your thoughts?

**Comment by harshit-splunk** [comment]
Issue is persistent with current master branch as well
```
gem 'kubeclient', git: 'https://github.com/ManageIQ/kubeclient.git', ref: 'a83fd436e1cddb295fba1a1bc02b7f1e0dbb43ec'
```

**Comment by cben** [comment]
Its not the builtin Net::HTTP, it sthe external [http gem](https://github.com/httprb/http)

**Comment by harshit-splunk** [comment]
Hi @cben, the issue is fixed in the HTTP gem is released https://rubygems.org/gems/http/versions/5.1.1
Can you release kubeclient gem with updated dependency when you get a chance?

**Comment by cben** [comment]
Oh great.   As it happens I just merged #589 that should allow http 5.  I indent to release later today.  Do you want to test v4.y branch to confirm it's fixed?

**Comment by harshit-splunk** [comment]
Unfortunately, my hands are full. I'll test it after the holidays.

**Comment by cben** [comment]
released 4.11.0.
